.github/workflows/snap.yaml

```diff
@@ -13,40 +13,184 @@ on:
 jobs:
 
   # Collect all relative paths containing a 'snap/snapcraft.yaml'
-  collect_dirs:
+  collect_snap_names:
     runs-on: ubuntu-latest
     outputs:
-      dirs: ${{ steps.dirs.outputs.dirs }}
+      snap-names: ${{ steps.snap-names.outputs.snap-names }}
     steps:
       - uses: actions/checkout@v3
-      - id: dirs
-        run: echo "dirs=$(find . -type f -name 'snapcraft.yaml' | sed -r 's|\/snap/[^/]+$||' |sort |uniq |jq --raw-input --slurp --compact-output 'split("\n")[:-1]')" >> $GITHUB_OUTPUT
+      - id: snap-names
+        run: echo "snap-names=$(find . -type f -name 'snapcraft.yaml' | sed -r 's|^\./([^/]+)/snap/snapcraft.yaml|\1|' | sed 's/-snap$//' | sort | uniq | jq --raw-input --slurp --compact-output 'split("\n")[:-1]')" >> $GITHUB_OUTPUT
 
   # Run snapcraft for each example
-  run_tests:
-    needs: collect_dirs
-    runs-on: ubuntu-latest
+  build_snaps:
+    needs: collect_snap_names
     strategy:
       fail-fast: false
       matrix:
-        dir: ${{ fromJson(needs.collect_dirs.outputs.dirs) }}
-    name: "Snap ${{ matrix.dir }}"
+        snap-name: ${{ fromJson(needs.collect_snap_names.outputs.snap-names) }}
+    name: "Snap ${{ matrix.snap-name }}"
+    uses: ubuntu-robotics/snap_workflows/.github/workflows/build-install-test-snap.yaml@F_add_option_to_select_snap_path
```

Comment:
this will change once that option is merged https://github.com/ubuntu-robotics/snap_workflows/pull/13#pullrequestreview-1844242649 , should we wait for it before merging this?

Reply:
yep